extensions/functions_string.yaml

```diff
@@ -163,3 +165,146 @@ scalar_functions:
           - value: "fixedchar<L1>"
           - value: "varchar<L2>"
         return: "BOOLEAN"
+  -
+    name: ltrim
+    description: >-
+      Remove any occurrence of the characters from the left side of the string.
+      If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+          - value: "varchar<L2>"
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+          - value: "string"
+        return: "string"
+      - args:
+          - value: "fixedchar<L1>"
+          - value: "fixedchar<L2>"
+        return: "string"
+  -
+    name: rtrim
+    description: >-
+      Remove any occurrence of the characters from the right side of the string.
+      If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+          - value: "varchar<L2>"
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+          - value: "string"
+        return: "string"
+      - args:
+          - value: "fixedchar<L1>"
+          - value: "fixedchar<L2>"
+        return: "string"
+  -
+    name: trim
+    description: >-
+      Remove any occurrence of the characters from the left and right sides of
+      the string. If no characters are specified, spaces are removed.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+          - value: "varchar<L2>"
+        return: "varchar<L1>"
+      - args:
+          - value: "string"
+          - value: "string"
+        return: "string"
+      - args:
+          - value: "fixedchar<L1>"
+          - value: "fixedchar<L2>"
+        return: "string"
+  -
+    name: lpad
+    description: Left-pad the string with the characters until the specified length of the string has been reached.
+    impls:
+      - args:
+          - value: "varchar<L1>"
+          - value: i32
+          - value: "varchar<L2>"
+        return: "string"
```

Comment:
The return for these padding functions kind of confused me.  I guess the return would be a varchar, where the length is the i32 input?

Reply:
You can't make a return type depend on an argument value. Imagine for instance a project relation that applies this function to three separate columns; what would the type of the resulting column be? You could do this if Substrait would support something akin to non-type template arguments in C++, but currently it doesn't.